### PR TITLE
add eliminate_nop_expand pass

### PR DESCRIPTION
**Pull request by HSQ79815**
Signed-off-by: haoshengqiang <haoshengqiang79@163.com>

### CODE REVIEW

File: onnxoptimizer/passes/eliminate_nop_expand.h [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      return false;
+
+    ndim_a--;
+    ndim_b--;
+
+    for (; ndim_a >= 0; ndim_a--, ndim_b--) {
+      int d_a = dims_a[ndim_a];
+      auto const& d_b = dims_b[ndim_b];
+      if (d_a == 1)
+        continue;
```
Comment (daquexian) [inline onnxoptimizer/passes/eliminate_nop_expand.h:45]: always wrap if body with {}, same for line 36

File: onnxoptimizer/passes/eliminate_nop_expand.h [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  explicit EliminateNopExpand()
+      : PredicateBasedPass(PassType::Nop, PassEfficiency::Complete,
+                           PassOptimizationType::Compute) {}
+
+  std::string getPassName() const override {
+    return "eliminate_nop_expand";
+  }
+
+  bool isConstantTensor(Graph& graph, const std::string& name) {
+    auto& initializer_names = graph.initializer_names();
```
Comment (daquexian) [inline onnxoptimizer/passes/eliminate_nop_expand.h:26]:
```suggestion
    const auto& initializer_names = graph.initializer_names();
```

**Review by daquexian — COMMENTED** [review]
Thanks for your great contribution! Could you please add a unit test for the new pass?

File: onnxoptimizer/passes/eliminate_nop_expand.h (hunk 1 above)
Comment (HSQ79815) [reply onnxoptimizer/passes/eliminate_nop_expand.h:45]: ok

**Comment by daquexian** [comment]
Great, thanks for your contribution!

**Pull request merged**